src/viewport/telemetry_safety.py

```
@@ -0,0 +1,326 @@
+    text = _PUBLIC_SHARE_ROUTE_RE.sub(r"\1{share_id}", text)
+    segments = text.split("/")
+    for index, segment in enumerate(segments):
+        if not _UUID_TEXT_RE.fullmatch(segment):
+            continue
+        previous = segments[index - 1].lower() if index > 0 else ""
+        segments[index] = "{" + _ROUTE_UUID_LABELS.get(previous, "uuid") + "}"
+    text = "/".join(segments)
+    return _UUID_SEGMENT_RE.sub("{uuid}", text)
+
+
+def redact_text(value: Any) -> str:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)